### DISCUSSION TITLE
Nutshell 🐿️ - real time meeting transcriptions powered by MLX-Whisper

### DISCUSSION THREAD

**Discussion by NutshellAI** [ml-explore/mlx#1573] [Show and tell]
Hi everyone! I’m excited to share **_Nutshell_**, a <ins>**completely private**</ins>, AI-powered transcription and meeting assistant application that leverages [MLX’s Whisper solution](https://github.com/ml-explore/mlx-examples/tree/main/whisper) for real-time, on-device transcriptions optimized for macOS. We've been working on it for the past couple of months and finally have a product that we're ready to share.

## **What is Nutshell?** 
Nutshell captures your meetings, transcribing audio and summarizing discussions, all while keeping your data secure on your device. This ensures full privacy and compliance with corporate data policies. 🎧✨

### Key Features:
- <ins>**Real-Time Transcription**</ins> powered by Whisper 
- <ins>**AI Meeting Assistant**</ins> that provides contextual summaries, action items, and Q&A
- <ins>**On-Device Processing**</ins> for enhanced privacy
- <ins>**Customizable Prompts**</ins> to tailor summaries and actions
- <ins>**Organizational Tools**</ins> with folders and export options for meeting transcriptions
- <ins>**No bots**</ins> ever sent to your meetings

**You can download and try it out on our website: [ withnutshell.com](https://withnutshell.com/). Let me know what you think, would love to hear your feedback!**

🙏 _A big thank you to the the MLX community, we are incredibly grateful open source communities like this one. Your shared knowledge, support, and collaborative spirit have been instrumental in our learning journey and growth while building Nutshell._ 

_One of the big hurdles that we faced with Nutshell was getting the app to run locally on Macbooks without needing crazy GPU upgrades. We wanted Nutshell to be accessible to everyone without needing special hardware, and MLX really helped make that happen._

❤️ 🐿️ 
from the founders at Nutshell

![main_nutshell](https://github.com/user-attachments/assets/9996ec6e-993a-43d6-95dd-82ac1713e666)

**Comment by awni** [comment]
The app looks very cool! I'm curious if you could say more about how you packaged MLX in it? Are you using MLX Whisper in Python inside the app or is it a custom Swift implementation or some other thing?

**Reply by NutshellAI** [reply]
Thanks! MLX whisper is packaged as a python executable and that is packaged as part of our app

**Comment by awni** [comment]
I posted about this on twitter: https://x.com/awnihannun/status/1854933494126068066

There are some questions in the thread that I don't know the answer to. Just FYI maybe you can answer or point them to a place they can get their answers.

**Reply by NutshellAI** [reply]
Wow! I really appreciate that! I'll respond to their question once I get back to my desk
